app/aws-lsp-codewhisperer-runtimes/scripts/create-repo-manifest.ts

```diff
@@ -0,0 +1,346 @@
+// Produces an artifact manifest that is "close enough" to a real one, to
+// reference artifacts attached to a GitHub release for main or feature branches.
+// The GitHub release only ever holds one artifact version (the most recent commit),
+// so the manifest will only ever contain a single artifact version.
+
+import * as fs from 'fs'
+import { exec } from 'child_process'
+import * as path from 'path'
+
+export type SemVer = string
+
+export type Version = {
+    name: string
+    version: SemVer
+}
+
+export type Platform = 'windows' | 'linux' | 'darwin'
+export type Arch = 'arm64' | 'x64'
+export type TargetContent = {
+    filename: string
+    url: string
+    hashes: string[]
+    bytes: number
+}
+
+export type PlatformTarget = {
+    platform: Platform
+    arch: Arch
+    contents: TargetContent[]
+}
+
+export type ManifestServerVersionEntry = {
+    serverVersion: string
+    isDelisted: boolean
+    runtime: Version
+    capabilities?: Version[]
+    protocol?: Version[]
+    thirdPartyLicenses: string
+    targets: PlatformTarget[]
+}
+
+export type Manifest = {
+    manifestSchemaVersion: string
+    artifactId: string
+    artifactDescription: string
+    isManifestDeprecated: boolean
+    versions: ManifestServerVersionEntry[]
+}
+
+interface Params {
+    serverVersion: string
+    releaseArtifactsPath: string
+    repoUrl: string
+    gitHubReleaseName: string
+}
+
+const serversZipName = 'servers.zip'
+const clientsZipName = 'clients.zip'
+
+/**
+ * Updates the manifest file with new version information or performs rollback operations.
+ *
+ * @async
+ * @param {string} manifestPath - file path to save the manifest to.
+ * @param {Object} params - The parameters for updating the manifest.
+ * @param {string} params.serverVersion - The server version.
+ * @param {string} params.releaseArtifactsPath - folder containing the artifacts to load file attributes from
+ * @param {string} params.repoUrl - url to the github repo (https://github.com/aws/language-servers)
+ * @param {string} params.gitHubReleaseName - the name of the GitHub release this will refer to (pre-main)
+ *
+ * @description
+ * This function performs the following operations:
+ * 1. Calculates SHA384 checksums and file sizes for clients.zip and servers.zip files.
+ * 2. Generates a new manifest entry with the provided and calculated information.
+ * 3. Produces a manifest file, containing only this one version.
+ * 4. Saves the updated manifest to a file.
+ */
+export async function updateManifest(
```

Comment:
if anything inside this function fails, does github CI show as aborted?

Reply:
The GitHub action will show as aborted on https://github.com/aws/language-servers/actions, however it will not be surfaced as an individual commit check.